test/org/javarosa/model/xform/SmsSerializingVisitorTest.java

```diff
@@ -0,0 +1,145 @@
+package org.javarosa.model.xform;
+
+import org.javarosa.core.PathConst;
+import org.javarosa.core.model.instance.FormInstance;
+import org.javarosa.core.services.transport.payload.ByteArrayPayload;
+import org.javarosa.core.test.FormParseInit;
+import org.javarosa.form.api.FormEntryController;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Performs a comprehensive suite of tests on the results of a form
+ * being serialized to the SMS format.
+ */
+public class SmsSerializingVisitorTest {
+    private String sms;
+    private static final String delimiter = " ";
+
+    @Before
+    public void setup() throws IOException {
```

Comment:
I know it’s rare to see a method spelled `setUp` (instead of setup), but `setUp` is correct because this method name is a verb, and it’s a phrasal verb made of two words. So we use camelcase. If it were a noun or adjective, it would be `setup`.